src/components/organisms/UiContentArea/UiContentArea.vue

```diff
@@ -0,0 +1,293 @@
+          v-bind="{
+            isActive,
+            buttonBackAttrs,
+            handleBackClick,
+          }"
+        >
+          <UiButton
+            v-if="isActive"
+            v-bind="buttonBackAttrs"
+            class="ui-button--icon ui-content-area__back"
+            @click="handleBackClick"
+          >
+            <UiIcon
+              v-bind="defaultProps.iconBackAttrs"
+              class="ui-button__icon"
+            />
+            <span class="visual-hidden">{{ defaultProps.translation.back }} {{ backToTitle }}</span>
+          </UiButton>
+        </slot>
+        <!-- @slot Use this slot to replace title template. -->
+        <slot
+          name="title"
+          v-bind="{
+            headingTitleAttrs,
+            title: currentTitle
+          }"
+        >
+          <UiHeading v-bind="headingTitleAttrs">
+            {{ currentTitle }}
+          </UiHeading>
+        </slot>
+      </div>
+    </slot>
+    <div class="ui-content-area__wrapper">
+      <div
+        :class="[
+          'ui-content-area__section', { 'ui-content-area__section--is-active': isActive }
+        ]"
+      >
+        <!-- @slot Use this slot to replace menu template. -->
+        <slot
+          name="menu"
+          v-bind="{ items: menuItems }"
+        >
+          <UiMenu
+            :items="menuItems"
+            class="ui-content-area__menu"
+          />
+        </slot>
+        <!-- @slot Use this slot to replace content template. -->
+        <slot name="content">
+          <div class="ui-content-area__content">
+            <!-- @slot Use this slot to place inside pages items. -->
+            <slot>
+              <template
+                v-for="(item, key) in items"
+                :key="key"
+              >
+                <UiContentAreaItem
+                  :label="item.label"
+                  :title="item.title"
+                  :name="item.name"
+                >
+                  <slot
+                    v-bind="{ item }"
+                    :name="item.name"
+                  />
+                </UiContentAreaItem>
+              </template>
+            </slot>
+          </div>
+        </slot>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import {
+  ref,
+  computed,
+  provide,
+  inject,
+} from 'vue';
+import type { PropType } from 'vue';
+import type { PropsAttrs } from '@/types/attrs';
+import type { Icon } from '../../../types/icon';
+import type { MenuSuffixVisible } from '../UiMenu/UiMenu.vue';
+import UiButton from '../../atoms/UiButton/UiButton.vue';
+import UiIcon from '../../atoms/UiIcon/UiIcon.vue';
+import UiHeading from '../../atoms/UiHeading/UiHeading.vue';
+import UiMenu from '../UiMenu/UiMenu.vue';
+import UiContentAreaItem from './_internal/UiContentAreaItem.vue';
+
+export interface ContentAreaItem {
+  label: string;
+  title: string;
+  name?: string;
+  [key: string]: unknown;
+}
+
+const props = defineProps({
```

Comment:
I wrote it already in [this PR](https://github.com/infermedica/component-library/pull/43), so you can dig into the discussion; 
**tldr** Why not use `withDefaults` instead of casting those props types over and over? 😜

Reply:
This way, unsupport docs comment and disallow to add props descriptions to the Storybook, but @ArkadiuszKrzymienski will change that.